### ISSUE TITLE
Mode indicator as toolbar button

### ISSUE THREAD

**Issue by SammysHP** [tridactyl/tridactyl#1511]
After being a user and developer of Vimperator for many years my layout of Firefox was just the web page and the statusbar, all Firefox UI completely hidden.

With Tridactyl not all features are available yet from within Tridactyl, so I kept the nav/toolbar. This might change with a complete statusbar, but I fear that webextension limitations will require at least some parts of the Firefox UI.

The mode indicator in the current form looks a bit strange and overlays parts of the web page. Because a complete statusbar is not yet available and because the navbar might be necessary in the future, **I suggest to add an alternative to the mode indicator in form of a toolbar button with different icon (and optionally color) for each mode.** Thus the mode indicator can be disabled via `:set modeindicator false` and a toolbar button added if desired.

- [ ] Add toolbar button for vimperator showing the current mode in form of an icon
- [ ] Optionally change the background color with the mode
- [ ] Optionally link the button to a page like Tridactyl preferences
- [ ] Optionally use the button's context menu to toggle some major features/modes like ignore mode

**Comment by quisquama** [comment]
Yes this would be great.  Currently, I use [theme that turns the status indicator into a little triangle](https://gist.github.com/quisquama/0f8ad71c24eccdc739e8e4a434452f2e) with different colors for different modes. But I'd prefer to have a button in the toolbar to indicate modes.

This feature would also be useful with the "superignore" mode discussed in this issue: https://github.com/tridactyl/tridactyl/issues/822#issuecomment-883461826
